supertokens_python/recipe/emailverification/api/implementation.py

```diff
@@ -13,65 +13,127 @@
+
+               - token: string,
+               - api_options: APIOptions,
+               - user_context: python dict
+
+            if the user is successfully varified, the response could be as follows
+
+                {
+                    "status": "OK",
+                    "user": {
+                      "id": "<usern-id>",
+                      "email": "<email-id>"
+                }
+
+            If the token is invalid and the user is not varified,
+            the response could be as follows:
+
+                {
+                    "status": "EMAIL_VERIFICATION_INVALID_TOKEN_ERROR"
+                }
+        """
+
         response = await api_options.recipe_implementation.verify_email_using_token(token, user_context)
-        if response.is_ok:
-            if response.user is None:
-                raise Exception("Should never come here")
+        if isinstance(response, VerifyEmailUsingTokenOkResult):
             return EmailVerifyPostOkResponse(response.user)
         return EmailVerifyPostInvalidTokenErrorResponse()
 
-    async def is_email_verified_get(self, api_options: APIOptions, user_context: Dict[str, Any]) -> IsEmailVerifiedGetResponse:
+    async def is_email_verified_get(self, api_options: APIOptions, user_context: Dict[str, Any]) -> IsEmailVerifiedGetOkResponse:
+        """
+        is_email_verified_get is an async method that accepts
+
+            - api_options: APIOptions
+            - user_context: Dict
+
+        returns a json/dict response as follows if the email is verified
+
+            {
+                "status": "OK",
+                "isVerified": true
+            }
+
+        Otherwise it raises an exception.
+        """
+
+        # TODO: fix docstring, docstring should mention exact functioning of this method.
+        # TODO: this method should return an error response instead of an exception.
+
         session = await get_session(api_options.request)
-        if session is None:
-            raise Exception('Session is undefined. Should not come here.')
 
-        user_id = session.get_user_id(user_context)
-        email = await api_options.config.get_email_for_user_id(user_id, user_context)
+        if isinstance(session, SessionContainer):
+            user_id = session.get_user_id(user_context)
+            email = await api_options.config.get_email_for_user_id(user_id, user_context)
+            is_verified = await api_options.recipe_implementation.is_email_verified(user_id, email, user_context)
+            return IsEmailVerifiedGetOkResponse(is_verified)
 
-        is_verified = await api_options.recipe_implementation.is_email_verified(user_id, email, user_context)
-        return IsEmailVerifiedGetOkResponse(is_verified)
+        raise Exception("Undefined Session")
 
     async def generate_email_verify_token_post(self, api_options: APIOptions, user_context: Dict[str, Any]) -> GenerateEmailVerifyTokenPostResponse:
```

Comment:
The return type of this function should also change to be a union of all the types that it actually returns. And then you don't need `GenerateEmailVerifyTokenPostResponse` anymore.